src/masternode.cpp

```diff
@@ -23,23 +23,31 @@ std::map<int64_t, uint256> mapCacheBlockHashes;
 //Get the last hash that matches the modulus given. Processed in reverse order
 bool GetBlockHash(uint256& hash, int nBlockHeight)
 {
-    if (chainActive.Tip() == NULL) return false;
+    int tipHeight;
+    const CBlockIndex* tipIndex;
+    {
+        LOCK(cs_main);
+        CBlockIndex *pindex = chainActive.Tip();
+        if (!pindex) return false;
+        tipHeight = pindex->nHeight;
```

Comment:
nit: we can initialize this after the `LOCK` code-block, as 
```cpp
int tipHeight = tipIndex->nHeight
```
(...or just remove it, and use `tipIndex->nHeight` directly).